File: app/src/main/java/com/meunicorn/fancy2u/UI/Activivty/MainActivity.java
package com.meunicorn.fancy2u.UI.Activivty;

import android.app.SearchManager;
import android.content.Context;
import android.os.Bundle;
import android.os.PersistableBundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.NavigationView;
import android.support.design.widget.TabLayout;
import android.support.v4.view.GravityCompat;
import android.support.v4.view.ViewPager;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import com.meunicorn.fancy2u.Bean.Shots.Shot;
import com.meunicorn.fancy2u.R;
import com.meunicorn.fancy2u.UI.Adapter.ShotFragmentViewPagerAdapter;
import com.meunicorn.fancy2u.UI.Fragment.ShotFragment;
import com.meunicorn.fancy2u.Utils.ActivityStart;

/**
 * Created by Fancy on 2016/4/27.
 */
public class MainActivity extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener, ShotFragment.OnListFragmentInteractionListener {
    DrawerLayout drawer;
    ViewPager viewPager;
    TabLayout tabLayout;

    @Override
    public void onListFragmentInteraction(Shot item) {
        ActivityStart.toDetail(this, item);
    }


    @Override
    public boolean onNavigationItemSelected(MenuItem item) {
        int id = item.getItemId();
        switch (id) {
            case R.id.nav_buckets:
                break;
            case R.id.nav_follwers:
                break;
            case R.id.nav_likes:
                break;
            case R.id.nav_projects:
                break;
            case R.id.nav_shots:
                break;
            case R.id.nav_teams:
                break;
            case R.id.nav_about:
                break;
            default:
                break;
        }
        drawer.closeDrawer(GravityCompat.START);
        return true;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ActivityStart.toLogin(MainActivity.this);
            }
        });
        drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
        viewPager= (ViewPager) findViewById(R.id.vp_main_display);
        tabLayout= (TabLayout) findViewById(R.id.tab_main_title);
        ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
        drawer.addDrawerListener(toggle);
        toggle.syncState();

        NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
        navigationView.setNavigationItemSelectedListener(this);
        initViewPager();
    }

    private void initViewPager() {
        ShotFragmentViewPagerAdapter adapter = new ShotFragmentViewPagerAdapter(getSupportFragmentManager());
        adapter.addFragment(new ShotFragment().newInstance("popularity"), getResources().getString(R.string.popular));
        adapter.addFragment(new ShotFragment().newInstance("recent"), getResources().getString(R.string.recent));
        adapter.addFragment(new ShotFragment().newInstance("comments"), getResources().getString(R.string.comments));
        adapter.addFragment(new ShotFragment().newInstance("views"), getResources().getString(R.string.views));
        viewPager.setAdapter(adapter);
        tabLayout.setupWithViewPager(viewPager);
    }

    @Override
    public void onBackPressed() {
        if (drawer.isDrawerOpen(GravityCompat.START)) {
            drawer.closeDrawer(GravityCompat.START);
        } else {
            super.onBackPressed();
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.main, menu);
        MenuItem searchItem = (MenuItem) findViewById(R.id.search_shot);
        SearchManager searchManager = (SearchManager) getSystemService(Context.SEARCH_SERVICE);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        int id = item.getItemId();
        if (id == R.id.action_settings) {
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: app/src/main/java/com/meunicorn/fancy2u/UI/Activivty/MainActivity.kt
package com.meunicorn.fancy2u.UI.Activivty

import android.net.Uri
import android.os.Bundle
import android.support.design.widget.FloatingActionButton
import android.support.design.widget.Snackbar
import android.view.View
import android.support.design.widget.NavigationView
import android.support.design.widget.TabLayout
import android.support.v4.view.GravityCompat
import android.support.v4.view.ViewPager
import android.support.v4.widget.DrawerLayout
import android.support.v7.app.ActionBarDrawerToggle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast
import com.meunicorn.fancy2u.Bean.Shots.Shot
import com.meunicorn.fancy2u.R
import com.meunicorn.fancy2u.UI.Adapter.ShotFragmentViewPagerAdapter
import com.meunicorn.fancy2u.UI.Fragment.DummyFragment
import com.meunicorn.fancy2u.UI.Fragment.ShotFragment
import com.meunicorn.fancy2u.UI.Fragment.dummy.DummyContent
import kotlinx.android.synthetic.main.app_bar_main.*
import kotlinx.android.synthetic.main.content_main.*

class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener ,DummyFragment.OnFragmentInteractionListener,ShotFragment.OnListFragmentInteractionListener{
    override fun onListFragmentInteraction(item: Shot) {
        Toast.makeText(this@MainActivity,item.title,Toast.LENGTH_SHORT).show()
    }

    override fun onFragmentInteraction(uri: Uri?) {
        throw UnsupportedOperationException()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val toolbar = findViewById(R.id.toolbar) as Toolbar
        setSupportActionBar(toolbar)

        val fab = findViewById(R.id.fab) as FloatingActionButton
        fab.setOnClickListener { view -> Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG).setAction("Action", null).show() }

        val drawer = findViewById(R.id.drawer_layout) as DrawerLayout
        val toggle = ActionBarDrawerToggle(
                this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close)
        drawer.addDrawerListener(toggle)
        toggle.syncState()

        val navigationView = findViewById(R.id.nav_view) as NavigationView
        navigationView.setNavigationItemSelectedListener(this)
        initViewPager()
    }

    private fun initViewPager() {
        val adapter:ShotFragmentViewPagerAdapter= ShotFragmentViewPagerAdapter(supportFragmentManager)
        adapter.addFragment(ShotFragment(),"newest")
        adapter.addFragment(DummyFragment(),"trending")
        vp_main_display.adapter=adapter
        tab_main_title.setupWithViewPager(vp_main_display)
    }

    override fun onBackPressed() {
        val drawer = findViewById(R.id.drawer_layout) as DrawerLayout
        if (drawer.isDrawerOpen(GravityCompat.START)) {
            drawer.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true
        }

        return super.onOptionsItemSelected(item)
    }

    @SuppressWarnings("StatementWithEmptyBody")
    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        // Handle navigation view item clicks here.
        val id = item.itemId

        if (id == R.id.nav_camera) {
            // Handle the camera action
        } else if (id == R.id.nav_gallery) {

        } else if (id == R.id.nav_slideshow) {

        } else if (id == R.id.nav_manage) {

        } else if (id == R.id.nav_share) {

        } else if (id == R.id.nav_send) {

        }

        val drawer = findViewById(R.id.drawer_layout) as DrawerLayout
        drawer.closeDrawer(GravityCompat.START)
        return true
    }
}
